spec/status_spec.rb

```diff
@@ -403,30 +403,42 @@
   end
 
   describe 'errors' do
-    it 'lets you add an error' do
-      status = make_status(state: :started)
-      status.add_error({'message' => 'blah', 'foo' => 2})
-      expect(status.errors).to eq [{'message' => 'blah', 'foo' => 2}]
+    let(:status) { make_status(state: :started) }
+
+    describe 'adding' do
+      it 'adds a simple hash error' do
+        status.add_error({'message' => 'blah', 'foo' => 2})
+        expect(status.errors).to eq [{'message' => 'blah', 'foo' => 2}]
+      end
+
+      it 'adds an arbitrary object as error' do
+        class Error < Object
+          def initialize
+            @foo = 'bar'
+          end
+        end
+        status.add_error(Error.new)
+        expect(status.errors).to eq [{"foo"=>"bar"}]
+      end
+
     end
 
     it 'lets you add an error and survives a manual reload' do
-      status = make_status(state: :started)
       status.add_error({'message' => 'blah', 'foo' => 2})
-      status = Jobba::Status.find(status.id)
-      expect(status.errors).to eq [{'message' => 'blah', 'foo' => 2}]
+      found = Jobba::Status.find(status.id)
+      expect(found.errors).to eq [{'message' => 'blah', 'foo' => 2}]
     end
 
     it 'lets you add multiple errors' do
-      status = make_status(state: :started)
       status.add_error("howdy")
       status.add_error(2)
       status.add_error([1,2,3])
       status.add_error(boo: 2)
       expect(status.errors).to eq ["howdy", 2, [1,2,3], {'boo' => 2}]
-      status = Jobba::Status.find(status.id)
-      expect(status.errors).to eq ["howdy", 2, [1,2,3], {'boo' => 2}]
-      status.reload!
-      expect(status.errors).to eq ["howdy", 2, [1,2,3], {'boo' => 2}]
+      found = Jobba::Status.find(status.id)
+      expect(found.errors).to eq ["howdy", 2, [1,2,3], {'boo' => 2}]
```

Comment:
wonder if this could be worded a bit better to reveal intention of these assertions?   the first expect is on a in-memory object and the second to test that what's in memory has been persisted right?  would it be better to just use a find_by (which uses a where, which guarantees that you're reading from the db)?  just asking.

Reply:
yep, the `reload!` isn't really doing anything since it was just found.  I'll rework it a bit